app/exerlog/lib/UI/calendar/widgets/logout_button.dart

```diff
@@ -24,7 +25,46 @@ class _LogoutButtonState extends State<LogoutButton> {
             setState(() {
               _isLoading = true;
             });
-            await FirebaseAuth.instance.signOut();
+            bool shouldLogout = false;
+            await showDialog(
+              context: context,
+              builder: (context) {
+                return AlertDialog(
```

Comment:
The Alert Dialog probably shouldn't be defined inside the logout button itself. The dialog itself should be a reusable widget. And we can then call it using a mixin, this way we can call this alert from anywhere in the app, instead of redefining it everytime